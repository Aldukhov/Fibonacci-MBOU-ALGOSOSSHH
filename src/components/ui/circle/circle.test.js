import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";
import { Circle } from './circle';
import { Button } from '../button/button';
import { ElementStates } from '../../../types/element-states';

test('Circle with letters', () => {
const { getByTestId } = render ( <Circle letter = 'a'/>)
const pElement = getByTestId('letter');

    expect(pElement).toBeInTheDocument();

    expect(pElement.textContent).toBe('a');
})

test('Circle without text', () => {
    const {getByTestId} = render (<Circle />);

    const pElement = getByTestId('letter');

    expect(pElement).toBeInTheDocument();

    expect(pElement.textContent).toBe('');
})

test('Circle with head', () => {
    const {getByTestId} = render (<Circle head = '1'/>);

    const head = getByTestId('head');

    expect(head).toBeInTheDocument();

    expect(head).toHaveTextContent('1');
})


test('Circle with React element head', () => {

    const {getByTestId} = render (<Circle head = {<Button/>}/>);

    const head = getByTestId('head');

    expect(head).toBeInTheDocument();

    if (React.isValidElement(head)) {
        expect(React.isValidElement(head.props.children)).toBeTruthy();
    }
})

test('Circle with tail', () => {
    const {getByTestId} = render (<Circle tail = '1'/>);

    const tail = getByTestId('tail');

    expect(tail).toBeInTheDocument();

    expect(tail).toHaveTextContent('1');
})


test('Circle with React element tail', () => {

    const {getByTestId} = render (<Circle tail = {<Button/>}/>);

    const tail = getByTestId('tail');

    expect(tail).toBeInTheDocument();

    if (React.isValidElement(tail)) {
        expect(React.isValidElement(tail.props.children)).toBeTruthy();
    }
})


test('Circle with index', () => {

    const {getByTestId} = render (<Circle index = {5}/>);

    const index = getByTestId('index');

    expect(index).toBeInTheDocument();

    expect(index).toHaveTextContent('5');
})

test('Circle with isSmall', () => {

    const {getByTestId} = render (<Circle isSmall = {true}/>);

    const isSmall = getByTestId('isSmall');

    expect(isSmall).toBeInTheDocument();

    expect(isSmall).toHaveClass('small');
})

test('Circle with state default', () => {

    const {getByTestId} = render (<Circle state = {ElementStates.Default}/>);

    const isSmall = getByTestId('isSmall');

    expect(isSmall).toBeInTheDocument();

    expect(isSmall).toHaveClass('default');
})

test('Circle with state changing', () => {

    const {getByTestId} = render (<Circle state = {ElementStates.Changing}/>);

    const isSmall = getByTestId('isSmall');

    expect(isSmall).toBeInTheDocument();

    expect(isSmall).toHaveClass('changing');
})

test('Circle with state modified', () => {

    const {getByTestId} = render (<Circle state = {ElementStates.Modified}/>);

    const isSmall = getByTestId('isSmall');

    expect(isSmall).toBeInTheDocument();

    expect(isSmall).toHaveClass('modified');
})