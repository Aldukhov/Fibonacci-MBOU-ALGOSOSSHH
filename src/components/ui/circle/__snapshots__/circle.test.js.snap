// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle with React element head 1`] = `
<div
  class="text text_type_input text_color_input mb-4 absolute head element"
  data-testid="head"
>
  <button
    class="text text_type_button text_color_primary button undefined false "
    data-testid="btn"
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</div>
`;

exports[`Circle with React element tail 1`] = `
<div
  class="text text_type_input text_color_input mt-4 absolute tail30 element"
  data-testid="tail"
>
  <button
    class="text text_type_button text_color_primary button undefined false "
    data-testid="btn"
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</div>
`;

exports[`Circle with head 1`] = `
<div
  class="text text_type_input text_color_input mb-4 absolute head string"
  data-testid="head"
>
  1
</div>
`;

exports[`Circle with index 1`] = `
<p
  class="text text_type_input text_color_input mt-4 absolute index"
  data-testid="index"
>
  5
</p>
`;

exports[`Circle with isSmall 1`] = `
<div
  class="circle  small default"
  data-testid="isSmall"
  seconddata-testid="isSmallTrue"
>
  <p
    class="text text_type_circle text_color_input letter"
    data-testid="letter"
  />
</div>
`;

exports[`Circle with letters 1`] = `
<p
  class="text text_type_circle text_color_input letter"
  data-testid="letter"
>
  a
</p>
`;

exports[`Circle with state changing 1`] = `
<div
  class="circle   changing"
  data-testid="isSmall"
  seconddata-testid="isBig"
>
  <p
    class="text text_type_circle text_color_input letter"
    data-testid="letter"
  />
</div>
`;

exports[`Circle with state default 1`] = `
<div
  class="circle   default"
  data-testid="isSmall"
  seconddata-testid="isBig"
>
  <p
    class="text text_type_circle text_color_input letter"
    data-testid="letter"
  />
</div>
`;

exports[`Circle with state modified 1`] = `
<div
  class="circle   modified"
  data-testid="isSmall"
  seconddata-testid="isBig"
>
  <p
    class="text text_type_circle text_color_input letter"
    data-testid="letter"
  />
</div>
`;

exports[`Circle with tail 1`] = `
<div
  class="text text_type_input text_color_input mt-4 absolute tail30 string"
  data-testid="tail"
>
  1
</div>
`;

exports[`Circle without text 1`] = `
<p
  class="text text_type_circle text_color_input letter"
  data-testid="letter"
/>
`;
